hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/ws/Constants.java

```diff
@@ -57,36 +49,27 @@
+            .put(ExecuteOperationResponse.class.getName(), ForwardCommand.class)
+            .put(DeployApplicationResponse.class.getName(), ForwardCommand.class)
+            .put(AddJdbcDriverResponse.class.getName(), ForwardCommand.class)
+            .put(AddDatasourceResponse.class.getName(), ForwardCommand.class)
+            .put(ExportJdrResponse.class.getName(), ForwardCommand.class)
+            .put(GenericErrorResponse.class.getName(), ForwardCommand.class);
 
     /**
      * These are the only valid commands that can come from UI clients.
      */
     ValidCommandsMap VALID_COMMANDS_FROM_UI = new ValidCommandsMap()
             .put(EchoCommand.REQUEST_CLASS.getName(), EchoCommand.class)
-            .put(DeployApplicationCommand.REQUEST_CLASS.getName(), DeployApplicationCommand.class)
-            .put(AddJdbcDriverCommand.REQUEST_CLASS.getName(), AddJdbcDriverCommand.class)
-            .put(AddDatasourceCommand.REQUEST_CLASS.getName(), AddDatasourceCommand.class)
-            .put(ExecuteOperationCommand.REQUEST_CLASS.getName(), ExecuteOperationCommand.class)
-            .put(ExportJdrCommand.REQUEST_CLASS.getName(), ExportJdrCommand.class)
-            .put(GenericErrorResponseCommand.REQUEST_CLASS.getName(), GenericErrorResponseCommand.class);
+            .put(DeployApplicationRequest.class.getName(), ForwardCommand.class)
+            .put(AddJdbcDriverRequest.class.getName(), ForwardCommand.class)
+            .put(AddDatasourceRequest.class.getName(), ForwardCommand.class)
+            .put(ExecuteOperationRequest.class.getName(), ForwardCommand.class)
+            .put(ExportJdrRequest.class.getName(), ForwardCommand.class)
+            .put(GenericErrorResponse.class.getName(), ForwardCommand.class);
 
     // QUEUES AND TOPICS
-    Endpoint DEST_FEED_EXECUTE_OP = new Endpoint(Type.QUEUE, "FeedExecuteOperation");
-    Endpoint DEST_FEED_DEPLOY_APPLICATION = new Endpoint(Type.QUEUE, "FeedDeployApplication");
-    Endpoint DEST_FEED_ADD_JDBC_DRIVER = new Endpoint(Type.QUEUE, "FeedAddJdbcDriver");
-    Endpoint DEST_FEED_ADD_DATASOURCE = new Endpoint(Type.QUEUE, "FeedAddDatasource");
-    Endpoint DEST_FEED_EXPORT_JDR = new Endpoint(Type.QUEUE, "ExportJDR");
-
-    Endpoint DEST_UICLIENT_EXECUTE_OP_RESPONSE = new Endpoint(Type.QUEUE, "UIClientExecuteOperationResponse");
-    Endpoint DEST_UICLIENT_DEPLOY_APPLICATION_RESPONSE = new Endpoint(Type.QUEUE, "UIDeployApplicationResponse");
-    Endpoint DEST_UICLIENT_ADD_JDBC_DRIVER_RESPONSE = new Endpoint(Type.QUEUE, "UIAddJdbcDriverResponse");
-    Endpoint DEST_UICLIENT_ADD_DATASOURCE_RESPONSE = new Endpoint(Type.QUEUE, "UIAddDatasourceResponse");
-    Endpoint DEST_UICLIENT_EXPORT_JDR_RESPONSE = new Endpoint(Type.QUEUE, "UIExportJDRResponse");
+    Endpoint FEED_COMMAND_QUEUE = new Endpoint(Type.QUEUE, "FeedCommandQueue");
+    Endpoint UI_COMMAND_QUEUE = new Endpoint(Type.QUEUE, "UiCommandQueue");
```

Comment:
Yes, this is exactly the point where I am not 100% sure if I am not oversimplifying.

> having different queues per different messages allowed each listener to listen to specific messages and thus each listener was able to do "its thing" specific for the type of message.

I was able to find just two distinct cases (that I kept): (1) Echo that does the job here in gateway and (2) all other requests and reponses that are just routed between bus and websockets. For this second case I introduced `ForwardCommand` that is the same for all and there is no need to write boiler plate Commands and Listeners.

I admit that (1) and (2) covers just the present needs, but as you say additional commands can still be added.

As for the number queues, I was considering to introduce one queue per feed, but I could not find enough technical necessity for that.